org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java

```diff
@@ -304,7 +306,7 @@ private void populateMenuItem(String mode, LaunchShortcutExtension ext, Menu men
 		LaunchShortcutAction action;
 		if (configuration != null) {
 			action = new LaunchShortcutAction(mode, ext, configuration);
-			action.setText(configuration.getName());
+			action.setText(NLS.bind(DebugUIMessages.LaunchShortcutAction_combineLaunchShortcutName, configuration.getName(), ext.getLabel()));
```

Comment:
As mentioned I don't think this is actually good, especially as it duplicates the information (the items are already grouped and have an icon according to the type) and the original name is not necessarily useful here especially as currently there is no way to hide the original one.

Some Shortcuts for example use `...` to show the usually open a new dialog, that would then look really odd (example `My Maven Run (Maven Build ...)`).

Reply:
Icons are often not readable, and some OS recommend not showing them on menus, so they're not strong enough to resolve the problem. Grouping is also not obvious as items are on the same level, if you see

- Java Application
- MyClass
- MyClassOnJava20
- JUnit Test
- MyClass (1)
- MyClassOnJava20 (1)

it doesn't make things obvious on a flat world. Also, screen readers work in a way that they would read the selected menu, not the whole list, so it would still be interesting to have the menu label relatively standalone so someone who only reads it has all the necessary information.
I don't think we will be able to skip the type being printed on the menu item if we want to ensure good accessibility.

About the `...` on the menu item, then should we show the launch config type label instead?

Reply:
Sure but why no one ever complained here:
![grafik](https://user-images.githubusercontent.com/1331477/232492367-5d920e95-e549-4e82-b40d-feb5c8b0afbe.png)

... it was never a problem but its much more "confusing" there as it shows usually much more items ... so I don't really get why now "screenreaders" are a concern here? I'm not an expert in that area but thing the RIA information might be enhanced here instead of trying to put everything in the label...